cmd/ethrex/cli.rs

```diff
@@ -1,198 +1,287 @@
+        help_heading = "Node options"
+    )]
+    pub network: Option<String>,
+    #[arg(long = "bootnodes", value_name = "BOOTNODE_LIST", value_delimiter = ',', num_args = 1.., help = "Comma separated enode URLs for P2P discovery bootstrap.", help_heading = "P2P options")]
+    pub bootnodes: Vec<Node>,
+    #[arg(
+        long = "datadir",
+        value_name = "DATABASE_DIRECTORY",
+        help = "If the datadir is the word `memory`, ethrex will use the InMemory Engine",
+        default_value = DEFAULT_DATADIR,
+        help = "Receives the name of the directory where the Database is located.",
+        long_help = "If the datadir is the word `memory`, ethrex will use the `InMemory Engine`.",
+        help_heading = "Node options"
+    )]
+    pub datadir: String,
+    #[arg(long = "syncmode", default_value = "full", value_name = "SYNC_MODE", value_parser = utils::parse_sync_mode, help = "The way in which the node will sync its state.", long_help = "Can be either \"full\" or \"snap\" with \"full\" as default value.", help_heading = "P2P options")]
+    pub syncmode: SyncMode,
+    #[arg(
+        long = "metrics.port",
+        value_name = "PROMETHEUS_METRICS_PORT",
+        help_heading = "Node options"
+    )]
+    pub metrics_port: Option<String>,
+    #[arg(
+        long = "dev",
+        action = ArgAction::SetTrue,
+        help = "Used to create blocks without requiring a Consensus Client",
+        long_help = "If set it will be considered as `true`. The Binary has to be built with the `dev` feature enabled.",
+        help_heading = "Node options"
+    )]
+    pub dev: bool,
+    #[arg(
+        long = "evm",
+        default_value = "revm",
+        value_name = "EVM_BACKEND",
+        help = "Has to be `levm` or `revm`",
+        value_parser = utils::parse_evm_engine,
+        help_heading = "Node options"
+    )]
+    pub evm: EvmEngine,
+}
+
+impl Default for Options {
+    fn default() -> Self {
+        Self {
+            http_addr: Default::default(),
+            http_port: Default::default(),
+            log_level: Level::INFO,
+            authrpc_addr: Default::default(),
+            authrpc_port: Default::default(),
+            authrpc_jwtsecret: Default::default(),
+            p2p_enabled: Default::default(),
+            p2p_addr: Default::default(),
+            p2p_port: Default::default(),
+            discovery_addr: Default::default(),
+            discovery_port: Default::default(),
+            network: Default::default(),
+            bootnodes: Default::default(),
+            datadir: Default::default(),
+            syncmode: Default::default(),
+            metrics_port: Default::default(),
+            dev: Default::default(),
+            evm: Default::default(),
+        }
+    }
+}
+
+#[cfg(feature = "based")]
+#[derive(ClapParser)]
+pub struct BasedOptions {
```

Comment:
Can't we make this a subcommand? Like `ethrex l2`? Thoughts?

Reply:
We can, it was the idea to do it sooner or later. We can do it in another PR if it suites you.

Reply:
Filed https://github.com/lambdaclass/ethrex/issues/2246.